### ISSUE TITLE
Does not fix permissions of directories such as /usr

### ISSUE THREAD

**Issue by Vogtinator** [openSUSE/permissions#32]
On current Tumbleweed, this happens:
```
180315b991c4:/ # chkstat --system --set
180315b991c4:/ # chown 1000:1000 /usr
180315b991c4:/ # chkstat --system --set
/usr/: on an insecure path - /usr has different non-root owner who could tamper with the file.
...
```
The result is that after running `chkstat`, `/usr` is still owned by `1000:1000`.

This is the backstory of https://github.com/openSUSE/permissions/pull/31, which only got into the bad codepath of reading uninitialized memory after complaining about `/usr` and children instead of fixing it.

**Comment by maltek** [comment]
Yes, this is expected.

We do not have an entry for `/usr` so we don't know what the permissions should be.

We do have entries for paths below `/usr` with *different* users (not `1000`). We warn and refuse to touch anything below such a directory, because the unprivileged user that owns an ancestor directory might've done any kind of shenanigans to files below it and we don't just want to paper over that and 'fix' the ownership of a file whose integrity we can't check.

We've already found potential privilege escalations with this warning, e.g. CVE-2019-3689. I expect there are more. They'll have to be fixed individually.

**Comment by Vogtinator** [comment]
> We do not have an entry for /usr so we don't know what the permissions should be.

There is one though, in `/etc/permissions`:

```
#
# root directories:
#

/                                                       root:root          755
/root/                                                  root:root          700
[...]
/usr/                                                   root:root          755
[...]
```

**Comment by maltek** [comment]
Ah, I see - good catch. Thanks, that's indeed a bug.

**Comment by maltek** [comment]
(The complexity of all the different cases makes my head hurt.)

So apparently, we decided that we never fix ownership unless the current owner is root (i.e. contents can be trusted).

However, we accidentally show the wrong error message for directories - the one for untrusted ancestors instead of the one for untrusted ownership.

**Comment by maltek** [comment]
So the bad error message has been addressed in 2816be048cfff4beb3f364500df0fecf9e8fc5ec, and the refusal to change ownership is by design.

Closing this.
